src/corecel/grid/Interpolator.hh

```diff
@@ -84,6 +86,53 @@ class Interpolator
     using YTraits_t = detail::InterpolatorTraits<YI, real_type>;
 };
 
+//---------------------------------------------------------------------------//
+/*!
+ * Interpolate using a cubic spline.
+ *
+ * Given a set of \f$ n \f$ data points \f$ (x_i, y_i) \f$ such that \f$ x_0 <
+ * x_1 < \dots < x_{n - 1} \f$, a cubic spline \f$ S(x) \f$ interpolating on
+ * the points is a piecewise polynomial function consisting of \f$ n - 1 \f$
+ * cubic polynomials \f$ S_i \f$ defined on \f$ [x_i, x_{i + 1}] \f$. The \f$
+ * S_i \f$ are joined at \f$ x_i \f$ such that both the first and second
+ * derivatives, \f$ S'_i \f$ and \f$ S''_i \f$, are continuous.
+ *
+ * The \f$ i^{\text{th}} \f$ piecewise polynomial \f$ S_i \f$ is given by:
+ * \f[
+   S_i(x) = a_0 + a_1(x - x_i) + a_2(x - x_i)^2 + a_3(x - x_i)^3,
+ * \f]
+ * where \f$ a_i \f$ are the polynomial coefficients, expressed in terms of the
+ * second derivatives as:
+ * \f{align}{
+   a_0 &= y_i \\
+   a_1 &= \frac{\Delta y_i}{\Delta x_i} - \frac{\Delta x_i}{6} \left[ S''_{i +
+          1} + 2 S''_{i} \right] \\
+   a_2 &= \frac{S''_i}{2} \\
+   a_3 &= \frac{1}{6 \Delta x_i} \left[ S''_{i + 1} - S''_i \right]
+ * \f}
+ */
+template<typename T = ::celeritas::real_type>
+class SplineInterpolator
+{
+  public:
+    //!@{
+    //! \name Type aliases
+    using real_type = T;
+    using Point = Array<T, 3>;
```

Comment:
Since this isn't R3, perhaps this should be a small `struct Spline { real_type x; real_type y; real_type y_dd; };`? (That lets you eliminate the local enum in `operator()` as well.)